### PR TITLE
Sync with AR

### PR DESCRIPTION
**Pull request by ocefpaf**
@marcelotrevisani this gets us a little closer to a working build. I would not worry about the other blas implementations, just focus on openblas for now. However, that is still failing locally for me. Putting this here in case you have any clues. This is the failures I'm getting:

```
/opt/conda/conda-bld/caffe_1561297095235/work/src/caffe/layers/window_data_layer.cpp: In member function 'virtual void caffe::WindowDataLayer<Dtype>::load_batch(caffe::Batch<Dtype>*)':
/opt/conda/conda-bld/caffe_1561297095235/work/src/caffe/layers/window_data_layer.cpp:293:42: error: 'CV_LOAD_IMAGE_COLOR' was not declared in this scope
         cv_img = cv::imread(image.first, CV_LOAD_IMAGE_COLOR);
                                          ^~~~~~~~~~~~~~~~~~~
/opt/conda/conda-bld/caffe_1561297095235/work/src/caffe/layers/window_data_layer.cpp:293:42: note: suggested alternative: 'CV_IS_IMAGE_HDR'
         cv_img = cv::imread(image.first, CV_LOAD_IMAGE_COLOR);
                                          ^~~~~~~~~~~~~~~~~~~
                                          CV_IS_IMAGE_HDR
make[2]: *** [src/caffe/CMakeFiles/caffe.dir/layers/window_data_layer.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [src/caffe/CMakeFiles/caffe.dir/all] Error 2
```

### CODE REVIEW

**Comment by marcelotrevisani** [comment]
usually, that is an error related to ``opencv 4``, which opencv version are you using?

**Comment by marcelotrevisani** [comment]
if you pin opencv to 3 that should be fine, I am going to clone your branch to see if it solves it

**Comment by ocefpaf** [comment]
> if you pin opencv to 3 that should be fine, I am going to clone your branch to see if it solves it

Let's try that. It is using the latest 4.0.1.

**Comment by marcelotrevisani** [comment]
should we remove the pinned hdf?

**Comment by ocefpaf** [comment]
> should we remove the pinned hdf?

We can't b/c of `h5py` :unamused:

**Comment by marcelotrevisani** [comment]
I think we should add to the ``build.sh`` the cmake options``MKL_INCLUDE_DIR`` and ``MKL_RT_LIBRARY``

**Comment by ocefpaf** [comment]
> I think we should add to the `build.sh` the cmake options`MKL_INCLUDE_DIR` and `MKL_RT_LIBRARY`

Like I mentioned in https://github.com/conda-forge/caffe-feedstock/pull/49#issue-290890829 I do not want to make this even more complicated by addressing all the blas implementations. I'm focusing on openblas now. Once we get that one working we can try MKL.

**Comment by marcelotrevisani** [comment]
Oh sorry, just saw your message now
I will revert my commit
Apologies

**Comment by ocefpaf** [comment]
> Oh sorry, just saw your message now

If you want to play with that feel free to open another PR with the commits here. I'm still trying things (old hdf5, boost, etc) and it make take a while to test mkl.

**Comment by marcelotrevisani** [comment]
> > Oh sorry, just saw your message now
> 
> If you want to play with that feel free to open another PR with the commits here. I'm still trying things (old hdf5, boost, etc) and it make take a while to test mkl.

I will test it locally then, sorry for that :(

**Comment by marcelotrevisani** [comment]
Is that correct?
```
- Found Boost 1.70.0 at $PREFIX/lib/cmake/Boost-1.70.0
--   Requested configuration: QUIET COMPONENTS python
-- Found boost_python 1.70.0 at $PREFIX/lib/cmake/boost_python-1.70.0
--   libboost_python37.dylib
-- Adding boost_python dependencies: headers
-- Boost 1.46 found.

```
it found boost 1.70 for python but it is appearing boost 1.46
for some reason, it seems that it is messing with the boost on the system

**Comment by ocefpaf** [comment]
> Is that correct?
> 
> ```
> - Found Boost 1.70.0 at $PREFIX/lib/cmake/Boost-1.70.0
> --   Requested configuration: QUIET COMPONENTS python
> -- Found boost_python 1.70.0 at $PREFIX/lib/cmake/boost_python-1.70.0
> --   libboost_python37.dylib
> -- Adding boost_python dependencies: headers
> -- Boost 1.46 found.
> ```
> 
> it found boost 1.70 for python but it is appearing boost 1.46
> for some reason, it seems that it is messing with the boost on the system

Nope. It does not look OK :-/

**Comment by marcelotrevisani** [comment]
to force boost you might try on cmake options
```
    -DBoost_NO_BOOST_CMAKE=TRUE \
    -DBoost_NO_SYSTEM_PATHS=TRUE \
    -DBOOST_ROOT:PATHNAME=$PREFIX \
    -DBoost_LIBRARY_DIRS:FILEPATH=${PREFIX}/lib \
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,82 +1,70 @@
...
+  detect_binary_files_with_prefix: False
 
 requirements:
   build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - boost
     - cmake
```
Comment (marcelotrevisani) [inline recipe/meta.yaml]: I think cmake should be in ``build`` right?

**Comment by ocefpaf** [comment]
> to force boost you might try on cmake options
> 
> ```
>     -DBoost_NO_BOOST_CMAKE=TRUE \
>     -DBoost_NO_SYSTEM_PATHS=TRUE \
>     -DBOOST_ROOT:PATHNAME=$PREFIX \
>     -DBoost_LIBRARY_DIRS:FILEPATH=${PREFIX}/lib \
> ```

It did not work locally. I'll try in the CIs as soon.

File: recipe/meta.yaml (hunk 1 above)
Comment (ocefpaf) [reply recipe/meta.yaml]: I'm not sure that makes any difference but it is worth trying.

**Comment by marcelotrevisani** [comment]
I am testing locally but I am seeing another error, but we might also add
```
      -DBoost_INCLUDE_DIRS="${PREFIX}/include" \
      -DBoost_LIBRARIES="${PREFIX}/lib" \
```

File: recipe/build.sh [hunk 2]
```diff
@@ -1,18 +1,43 @@
...
+    BLAS=mkl
 fi
-cmake -DCPU_ONLY=1 -DBLAS="open" -DCMAKE_INSTALL_PREFIX="${PREFIX}" -DCMAKE_INSTALL_LIBDIR=lib -Dpython_version=$PY_VER ..
-make
-make runtest
+
+cmake -D CPU_ONLY=1 \
+      -D BLAS="${BLAS}" \
+      -D CMAKE_INSTALL_PREFIX="${PREFIX}" \
+      -D NUMPY_INCLUDE_DIR="${SITE_PKGS}/numpy/core/include" \
```
Comment (marcelotrevisani) [inline recipe/build.sh]: is that correct? or should it be ``SP_DIR`` instead of ``SITE_PKGS``?

Comment (ocefpaf) [reply recipe/build.sh]: The conda-build env var is SP_DIR but this was working in the AR recipe :-/

**Comment by marcelotrevisani** [comment]
I think the CMakeLists from Caffe is not well developed.
Because of that, I will create a Makefile from scratch based on their example
CMake is mixing system libs with conda

**Comment by ocefpaf** [comment]
> I think the CMakeLists from Caffe is not well developed.
> Because of that, I will create a Makefile from scratch based on their example
> CMake is mixing system libs with conda

@isuruf is the resident cmake wizard. Can you take a quick look at this @isuruf and see if we have an easy way out?

**Comment by marcelotrevisani** [comment]
@conda-forge-admin , please rerender

**Comment by marcelotrevisani** [comment]
@conda-forge-admin , please rerender

**Comment by marcelotrevisani** [comment]
@ocefpaf , could you please review it?

**Comment by ocefpaf** [comment]
:tada: 

Amazing work @marcelotrevisani!

**Pull request merged**